### DISCUSSION TITLE
Nested object's method `rules` is ignored when parent object is validated

### DISCUSSION THREAD

**Discussion by alexrififi** [spatie/laravel-data#134] [Q&A] [answered]
Structure like [this](https://spatie.be/docs/laravel-data/v1/as-a-data-transfer-object/nesting):

```php
class ArtistData extends Data
{
    public function __construct(
        public string $name,
        public int $age,
    ) {
    }
}

class AlbumData extends Data
{
    public function __construct(
        public string $title,
        public ArtistData $artist,
    ) {
    }
}
```

When validating an `AlbumData` object and `ArtistData` has method `rule`, then the rules specified in it are ignored 😕

**Comment by rubenvanassche** [comment, marked as answer]
Hi @medvinator,

Could you provide me a test which shows where this is going wrong? We have a test for nested data objects with overwritten rules : https://github.com/spatie/laravel-data/blob/main/tests/Resolvers/DataValidationRulesResolverTest.php#L74-L90 and it seems to be working fine to me.

**Reply by alexrififi** [reply]
Excuse me generously, Baron @rubenvanassche !
It was my mistake!
